jina/executors/indexers/dbms/keyvalue.py

```diff
@@ -0,0 +1,271 @@
+        """
+
+        # ! a || ( a && b )
+        # =
+        # ! a || b
+        if not self.handler_mutex or not self.is_handler_loaded:
+            r = self.get_add_handler() if self.is_exist else self.get_create_handler()
+
+            if r is None:
+                self.logger.warning(
+                    '"write_handler" is None, you may not add data to this index, '
+                    'unless "write_handler" is later assigned with a meaningful value'
+                )
+            else:
+                self.is_handler_loaded = True
+            return r
+
+    @property
+    def size(self) -> int:
+        """
+        The number of vectors or documents indexed.
+
+        :return: size
+        """
+        return self._size
+
+    def __getstate__(self):
+        d = super().__getstate__()
+        self.flush()
+        return d
+
+    @property
+    def default_dump_path(self):
+        return os.path.join(self.workspace, 'dump')
+
+    def close(self):
+        """Close all file-handlers and release all resources. """
+        self.logger.info(
+            f'indexer size: {self.size} physical size: {get_readable_size(physical_size(self.workspace))}'
+        )
+        self.flush()
+        if self._dump_on_exit:
+            shutil.rmtree(self.default_dump_path, ignore_errors=True)
+            self.dump({'dump_path': self.default_dump_path})
+        call_obj_fn(self.write_handler, 'close')
+        call_obj_fn(self.query_handler, 'close')
+        super().close()
+
+    def flush(self):
+        """Flush all buffered data to ``index_abspath`` """
+        try:
+            # It may have already been closed by the Pea using context manager
+            call_obj_fn(self.write_handler, 'flush')
+        except:
+            pass
+
+    def _filter_nonexistent_keys_values(
+        self, keys: Iterable, values: Iterable, existent_keys: Iterable
+    ) -> Tuple[Iterable, Iterable]:
+        f = [(key, value) for key, value in zip(keys, values) if key in existent_keys]
+        if f:
+            return zip(*f)
+        else:
+            return None, None
+
+    def _filter_nonexistent_keys(
+        self, keys: Iterable, existent_keys: Iterable
+    ) -> Iterable:
+        return [key for key in keys if key in set(existent_keys)]
+
+    @property
+    def index_abspath(self) -> str:
+        """
+        Get the file path of the index storage
+
+        :return: absolute path
+        """
+        os.makedirs(self.workspace, exist_ok=True)
+        return os.path.join(self.workspace, self.index_filename)
+
+    def _get_generator(
+        self, ids: List[str]
+    ) -> Generator[Tuple[str, np.array, bytes], None, None]:
+        for id_ in ids:
+            vecs_metas_bytes = super()._query(id_)
+            vec, meta = pickle.loads(vecs_metas_bytes)
+            yield id_, vec, meta
+
+    @requests(on='/dump')
+    def dump(self, parameters, *args, **kwargs) -> None:
```

Comment:
this is wrong, arguments are passed in order. see https://github.com/jina-ai/jina/blob/devel-2.0/COOKBOOK.md#method-arguments